src/java.management.rmi/share/classes/javax/management/remote/rmi/RMIConnector.java

```diff
@@ -623,14 +606,15 @@ private Integer[] addListenersWithSubjects(ObjectName[]       names,
     // Implementation of MBeanServerConnection
     //--------------------------------------------------------------------
     private class RemoteMBeanServerConnection implements MBeanServerConnection {
-        private Subject delegationSubject;
 
         public RemoteMBeanServerConnection() {
             this(null);
         }
 
         public RemoteMBeanServerConnection(Subject delegationSubject) {
-            this.delegationSubject = delegationSubject;
+            if (delegationSubject != null) {
+                throw new SecurityException("Subject Delegation has been removed.");
+            }
         }
```

Comment:
Can't this constructor be removed? It's a private inner class so we should be able to throw before calling that constructor?

Reply:
Yes that can disappear with no impact.